sql/src/main/java/io/crate/metadata/blob/BlobSchemaInfo.java

```diff
@@ -112,12 +112,12 @@ public void update(ClusterChangedEvent event) {
     }
 
     @Override
-    public Iterator<TableInfo> iterator() {
+    public Iterable<TableInfo> getTables() {
         return Stream.of(clusterService.state().metaData().getConcreteAllOpenIndices())
             .filter(BlobIndex::isBlobIndex)
             .map(BlobIndex::stripPrefix)
             .map(tableInfoFunction)
-            .iterator();
+            .collect(Collectors.toList());
```

Comment:
This could be kept lazy by using a `::iterator` method ref.
```
        Iterable<TableInfo> iterator = Stream.of(clusterService.state().metaData().getConcreteAllOpenIndices())
            .filter(BlobIndex::isBlobIndex)
            .map(BlobIndex::stripPrefix)
            .map(tableInfoFunction)
            ::iterator;
```
This avoids the unnecessary list creation.